hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/AbstractCollectionStreamSerializer.java

```diff
@@ -26,17 +26,15 @@
 /**
  * The {@link Collection} serializer
  */
-abstract class AbstractCollectionStreamSerializer<CollectionType extends Collection>
+abstract class AbstractCollectionStreamSerializer<CollectionType extends Collection<?>>
         implements StreamSerializer<CollectionType> {
 
     @Override
     public void write(ObjectDataOutput out, CollectionType collection) throws IOException {
         int size = collection.size();
         out.writeInt(size);
-        if (size > 0) {
-            for (Object o : collection) {
-                out.writeObject(o);
-            }
+        for (Object o : collection) {
```

Comment:
What is the reason to get rid of the if check? In theory scalar replacement could be applied to the iterator; preventing it to be created on the heap. But I'm not sure how well the JIT deal with this; especially with JDK 8.

Reply:
I consider it to be a bad style. Googled a bit and found this claim for OpenJDK Hotspo 1.8.0_121:
>Practically speaking, simple iterators will usually be stack allocatable if you just use them to iterate.

Reply:
Good to know.